Core/src/Material/Interactions.cpp

```diff
@@ -367,51 +374,57 @@ inline float deriveMuonDirectPairPhotoNuclearLossMeanE(double energy) {
 
 }  // namespace
 
-float Acts::computeEnergyLossRadiative(const MaterialSlab& slab, int pdg,
-                                       float m, float qOverP, float q) {
-  ASSERT_INPUTS(m, qOverP, q)
+float Acts::computeEnergyLossRadiative(const MaterialSlab& slab,
+                                       PdgParticle absPdg, float m,
+                                       float qOverP, float absQ) {
+  assert((absPdg == Acts::makeAbsolutePdgParticle(absPdg)) &&
+         "pdg is not absolute");
 
   // return early in case of vacuum or zero thickness
   if (not slab) {
     return 0.0f;
   }
 
   // relative radiation length
-  const auto x = slab.thicknessInX0();
+  const float x = slab.thicknessInX0();
   // particle momentum and energy
   // do not need to care about the sign since it is only used squared
-  const auto momentum = q / qOverP;
-  const auto energy = std::hypot(m, momentum);
+  const float momentum = absQ / qOverP;
+  const float energy = std::hypot(m, momentum);
+
+  float dEdx = computeBremsstrahlungLossMean(m, energy);
 
-  auto dEdx = computeBremsstrahlungLossMean(m, energy);
-  if (((pdg == PdgParticle::eMuon) or (pdg == PdgParticle::eAntiMuon)) and
-      (8_GeV < energy)) {
+  // muon- or muon+
+  if ((absPdg == PdgParticle::eMuon) and (8_GeV < energy)) {
```

Comment:
Do we actually know where the 8 GeV come from? 😄

Reply:
I don't :smile:

Reply:
added a `TODO`